seed/challenges/01-front-end-development-certification/basic-javascript.json

```diff
@@ -31,66 +32,36 @@
+        "In computer science, <code>data</code> is anything that is meaningful to the computer. JavaScript provides seven different <dfn>data types</dfn> which are <code>undefined</code>, <code>null</code>, <code>Boolean</code>, <code>string</code>, <code>symbol</code>, <code>number</code>, and <code>object</code>.",
+        "For example, computers distinguish between numbers, such as the number <code>12</code>, and <code>strings</code>, such as <code>\"12\"</code>, <code>\"dog\"</code>, or <code>\"123 cats\"</code>, which are collections of characters. Computers can perform mathematical operations on a number, but not on a string.",
+        "<code>Variables</code> allow computers to store data in a dynamic fashion, rather than updating a formula every time the data changes. Any of the seven different data types may be stored in a variable.",
+        "<code>Variables</code> are similar to the x and y variables you use in mathematics, which means they're a simple name to represent the data we want to refer to. Computer <code>variables</code> differ from mathematical variables in that they can store different values at different times.",
+        "We tell JavaScript to create or <dfn>declare</dfn> a variable by putting the keyword <code>var</code> in front of it, like so:",
+        "<blockquote>var ourName;</blockquote>",
+        "creates a <code>variable</code> called <code>ourName</code>.  In JavaScript we end statements with semicolons.",
+        "<code>Variable</code> names can be made up of numbers, letters, and <code>$</code> or <code>_</code>, but may not contain spaces or start with a number.",
         "<h4>Instructions</h4>",
         "Use the <code>var</code> keyword to create a variable called <code>myName</code>. Set its value to your name, wrapped in double quotes.",
-        "<strong>Hint</strong>",
-        "Look at the <code>ourName</code> example if you get stuck."
+        "<strong>Hint</strong><br>Look at the <code>ourName</code> example if you get stuck."
       ],
       "challengeSeed": [
         "// Example",
-        "var ourName = \"Free Code Camp\";",
+        "var ourName;",
         "",
-        "// Only change code below this line",
+        "// Define myName below this line",
         ""
       ],
       "tail": [
         "if(typeof(myName) !== \"undefined\"){(function(v){return v;})(myName);}"
       ],
       "solutions": [
-        "var myName = \"Test Testerson\";"
+        "var myName;"
       ],
       "tests": [
-        "assert((function(){if(typeof(myName) !== \"undefined\" && typeof(myName) === \"string\" && myName.length > 0){return true;}else{return false;}})(), 'message: <code>myName</code> should be a string that contains at least one character in it.');"
+        "assert(/var\\s+myName\\s*;/.test(code), 'message: You should declare <code>myName</code> with the <code>var</code> keyword, ending with a semicolon');"
```

Comment:
We have generally not had periods at the end of test messages since they are largely fragments.